bigquery_etl/cli/query.py

```diff
@@ -1310,29 +1341,43 @@ def initialize(name, sql_dir, project_id, dry_run):
     for query_file in query_files:
         sql_content = query_file.read_text()
 
-        # Enable init from query.sql files
-        # First deploys the schema, then runs the init
-        # This does not currently verify the accuracy of the schema
+        # Enable initialization from query.sql files
+        # Create the table by deploying the schema and metadata, then run the init.
+        # This does not currently verify the accuracy of the schema or that it
+        # matches the query.
         if "is_init()" in sql_content:
             project = query_file.parent.parent.parent.name
             dataset = query_file.parent.parent.name
             destination_table = query_file.parent.name
-            Schema.from_schema_file(query_file.parent / SCHEMA_FILE).deploy(
-                f"{project}.{dataset}.{destination_table}"
-            )
+            full_table_id = f"{project}.{dataset}.{destination_table}"
+
+            try:
+                client.get_table(full_table_id)
+                raise PreconditionFailed(
+                    f"Table {full_table_id} already exists. The initialization process is terminated."
+                )
```

Comment:
Requiring that the table not exist at all is overly strict IMO (e.g. the table may have already been created via the `bqetl_artifact_deployment.publish_new_tables` task).  I'd suggest allowing an empty table and/or having a `--force` option to override this.
```suggestion
                table = client.get_table(full_table_id)
                if table.num_rows > 0:
                    raise PreconditionFailed(
                        f"Table {full_table_id} already exists and contains data. The initialization process is terminated."
                    )
```
Also, using the BigQuery client's own `PreconditionFailed` exception for this purpose seems a bit odd to me.